lib/Common/Memory/PageAllocator.cpp

```diff
@@ -561,23 +571,47 @@ PageSegmentBase<T>::DecommitFreePages(size_t pageToDecommit)
     Assert(pageToDecommit != 0);
     char * currentAddress = this->address;
 
-    uint decommitCount = 0;
+    uint decommitCount = 0, lastDecommitCount = 0;
+    bool freeBlock = false;
     for (uint i = 0; i < this->GetAvailablePageCount(); i++)
     {
         if (this->TestInFreePagesBitVector(i))
```

Comment:
>TestInFreePagesBitVector [](start = 18, length = 24)

I think it would be good implement a method in the BitVector to get the consecutive bit count from a start point, to make it more generic.